go-controller/pkg/node/iptables/iptables_manager.go

```diff
@@ -304,36 +295,24 @@ func (c *Controller) reconcile() error {
 		klog.V(5).Infof("Reconciling IPTables rules took %v", time.Since(start))
 	}()
 
-	existingChainRulesV4 := make(map[Chain][]RuleArg)
-	existingChainRulesV6 := make(map[Chain][]RuleArg)
-
-	// Gather existing rules from tables and chains
-	for chain := range c.chainRules {
-		if chain.Proto == iptables.ProtocolIPv4 {
-			rules, err := c.GetIPv4ChainRuleArgs(chain.Table, chain.Chain)
-			if err != nil {
-				return fmt.Errorf("failed to find IPv4 rules for chain %s in table %s", chain.Chain, chain.Table)
-			} else {
-				existingChainRulesV4[chain] = rules
-			}
-		} else if chain.Proto == iptables.ProtocolIPv6 {
-			rules, err := c.GetIPv6ChainRuleArgs(chain.Table, chain.Chain)
-			if err != nil {
-				return fmt.Errorf("failed to find IPv6 rules for chain %s in table %s", chain.Chain, chain.Table)
-			} else {
-				existingChainRulesV6[chain] = rules
-			}
+	existingChainRules := make(map[rulesIndex]rules)
+	// Gather existing rules from tables and chains that we care about
+	for ruleIndex := range c.store {
```

Comment:
cont you need to lock the mutex to access c.store? Or if not because it is expected that callers have already locked it, then you should add a comment to the function saying must be called with controller locked.

Reply:
ya, mutex must be held by caller. I added a comment to say this is the case.